src/utils/server-utils.ts

```diff
@@ -52,7 +52,7 @@ export const getRequestDate = (
     date => date.displayDate === selectedDate,
   );
   if (!dateItem) {
-    return selectedDate;
+    return layerAvailableDates[layerAvailableDates.length - 1].queryDate;
```

Comment:
@JorgeMartinezG what do you think of this change?

Also, in this case `if (!layerAvailableDates)`, should we return `undefined` instead?